tests/test_mathematical.py

```diff
@@ -1042,23 +1044,46 @@ def test_sum_empty_out(dtype):
 
 
 @pytest.mark.parametrize(
-    "shape", [(), (1, 2, 3), (1, 0, 2), (10), (3, 3, 3), (5, 5), (0, 6)]
+    "shape",
+    [
+        (),
+        (1, 2, 3),
+        (1, 0, 2),
+        (10,),
+        (3, 3, 3),
+        (5, 5),
+        (0, 6),
+        (10, 1),
+        (1, 10),
+    ],
 )
 @pytest.mark.parametrize(
     "dtype_in", get_all_dtypes(no_complex=True, no_bool=True)
 )
 @pytest.mark.parametrize(
     "dtype_out", get_all_dtypes(no_complex=True, no_bool=True)
```

Comment:
Can we enable the test for bool and complex dtypes also?

Reply:
done.

Now in total there are about about 1300 tests on sum, which takes in total ~25 seconds.